apps/homescreen/js/firstrun.js

```diff
@@ -24,7 +24,7 @@
             var index = 0;
             json.grid.forEach((section) => {
               section.forEach((entry) => {
-                if (!entry.manifestURL) {
+                if (!entry || !entry.manifestURL) {
```

Comment:
We should unit test this, it already unit tests for null and empty JSON, but it doesn't test for null entries.